##  Milestone 2 - Project 2

**Name: Denilson Wilson Aguilar Molina**

**Date: 04/16/2024**

In [2]:
#Importing pandas
import pandas as pd

Viewing from the Excel file, the given dataset that I am working with is rather large as it contains 1000 rows of different imdb movies, containing 16 columns in total. The headers for each colum follow as Poster_Link, Series_Title, Released_Year, Certificate, Runtime, Genre, IMDB_Rating, Overview, Meta_score, Director, Star1, Star2, Star3, Star4, No_of_Votes, Gross.

This is where the dataset is read into a pandas dataframe.

In [4]:
df = pd.read_csv('imdb_movies.csv')

I will be following the provided order in the given part of this project for how I go about using the basic properties of a dataset.

I will start off using describe()

In [9]:
#Included this print to help illustrate that the results below are from describe()
print("Results:")

print(df.describe())

Results:
       IMDB_Rating  Meta_score   No_of_Votes
count  1000.000000  843.000000  1.000000e+03
mean      7.949300   77.971530  2.736929e+05
std       0.275491   12.376099  3.273727e+05
min       7.600000   28.000000  2.508800e+04
25%       7.700000   70.000000  5.552625e+04
50%       7.900000   79.000000  1.385485e+05
75%       8.100000   87.000000  3.741612e+05
max       9.300000  100.000000  2.343110e+06


This appears to be only refering to the IMDB ratings, meta score, and Number of votes. 

The average IMDB rating of movies is be 7.95 with a minimum of 7.6 and a maximum of 9.3, These ratings have a standard deviation of 0.28. 

The average meta score of movies is be 77.97 with a minimum of 28 and a maximum of 100, These ratings have a standard deviation of 12.38. 

The average number of votes for movies would be 273,692 with a minimum of 25,088 and a maximum of 2,343,110, These ratings have a standard deviation of 327,373. 


Below I'll be using columns.

In [10]:
print("Columns from the IMBD Movies dataset:")
print(df.columns)

Columns from the IMBD Movies dataset:
Index(['Poster_Link', 'Series_Title', 'Released_Year', 'Certificate',
       'Runtime', 'Genre', 'IMDB_Rating', 'Overview', 'Meta_score', 'Director',
       'Star1', 'Star2', 'Star3', 'Star4', 'No_of_Votes', 'Gross'],
      dtype='object')


The provided columns are exactly as the one's I was able to view from looking through the excel file itself.

Below I'll be describing what each one refers to:
"Poster_Link" refers to a link to the poster of the series/movie.
"Series_Title" refers the title of the series/movie.
"Released_Year" refers to the year the series/movie was released.
"Certificate" refers to the certification or rating of the series/movie (e.g., PG-13, R, etc.).
"Runtime" refers to the duration of the series/movie.
"Genre" refers to the category or genre of the series/movie.
"IMDB_Rating" refers to the Rating of the series/movie on IMDb.
"Overview" refers to a brief overview or summary of the series/movie.
"Meta_score" refers to the metascore of the series/movie.
"Director" refers to the director/s of the series/movie.
"Star1, Star2, Star3, Star4" refers to the lead actors or stars in the series/movie.
"No_of_Votes" refers to the number of votes the series/movie has received.
"Gross" refers to the gross revenue or earnings of the series/movie.

I'll be using shape below

In [11]:
print("dataset shape:", df.shape)

dataset shape: (1000, 16)


This shows that the datset itself does contain 1000 rows and 16 columns, each row represents a different items that under the given columns, each column represents a different attribute.

I will now be using dtypes

In [12]:
print("data types of columns:")
print(df.dtypes)

data types of columns:
Poster_Link       object
Series_Title      object
Released_Year     object
Certificate       object
Runtime           object
Genre             object
IMDB_Rating      float64
Overview          object
Meta_score       float64
Director          object
Star1             object
Star2             object
Star3             object
Star4             object
No_of_Votes        int64
Gross             object
dtype: object


What can be gathered from this is that columns such as "Poster_Link", "Series_Title", "Released_Year", "Certificate", "Runtime", "Genre", "Overview", "Director", "Star1", "Star2", "Star3", "Star4", and "Gross" are object type. 
Columns such as "IMDB_Rating" and "Meta_score" are float64 type.
The column "No_of_Votes" is the int64 type.

Below the head() will be used

In [17]:
print("First rows from the movie dataset:")
print(df.head())

First rows from the movie dataset:
                                         Poster_Link  \
0  https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1  https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2  https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3  https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4  https://m.media-amazon.com/images/M/MV5BMWU4N2...   

               Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  \
0                 Drama          9.3   
1          Crime, Drama          9.2   
2  Action, Crime, Drama          9.0   
3          Crime, Drama          9.0   
4          Crime, Drama        

What can be seen are the top few rows of the movie dataset, along with the information corresponding to each one.

Below I'll be using tail()

In [18]:
print("Last rows of movie dataset:")
print(df.tail())

Last rows of movie dataset:
                                           Poster_Link  \
995  https://m.media-amazon.com/images/M/MV5BNGEwMT...   
996  https://m.media-amazon.com/images/M/MV5BODk3Yj...   
997  https://m.media-amazon.com/images/M/MV5BM2U3Yz...   
998  https://m.media-amazon.com/images/M/MV5BZTBmMj...   
999  https://m.media-amazon.com/images/M/MV5BMTY5OD...   

               Series_Title Released_Year Certificate  Runtime  \
995  Breakfast at Tiffany's          1961           A  115 min   
996                   Giant          1956           G  201 min   
997   From Here to Eternity          1953      Passed  118 min   
998                Lifeboat          1944         NaN   97 min   
999            The 39 Steps          1935         NaN   86 min   

                        Genre  IMDB_Rating  \
995    Comedy, Drama, Romance          7.6   
996            Drama, Western          7.6   
997       Drama, Romance, War          7.6   
998                Drama, War          7.6

This shows the last few rows of the movies datasetm providing a glimpse of the entries at the end. 

I'll be using sample() below

In [15]:
print("Random sample of the dataset:")
print(df.sample())

Random sample of the dataset:
                                           Poster_Link  Series_Title  \
845  https://m.media-amazon.com/images/M/MV5BYmVhMD...  Le locataire   

    Released_Year Certificate  Runtime            Genre  IMDB_Rating  \
845          1976           R  126 min  Drama, Thriller          7.7   

                                              Overview  Meta_score  \
845  A bureaucrat rents a Paris apartment where he ...        71.0   

           Director           Star1            Star2           Star3  \
845  Roman Polanski  Roman Polanski  Isabelle Adjani  Melvyn Douglas   

            Star4  No_of_Votes      Gross  
845  Jo Van Fleet        39889  1,924,733  


This shows a raandom sample from the dataset, it randomly selects rows and prints it out. 

Finally I'll be using info()

In [16]:
print("Info from dataset:")
print(df.info())

Info from dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Poster_Link    1000 non-null   object 
 1   Series_Title   1000 non-null   object 
 2   Released_Year  1000 non-null   object 
 3   Certificate    899 non-null    object 
 4   Runtime        1000 non-null   object 
 5   Genre          1000 non-null   object 
 6   IMDB_Rating    1000 non-null   float64
 7   Overview       1000 non-null   object 
 8   Meta_score     843 non-null    float64
 9   Director       1000 non-null   object 
 10  Star1          1000 non-null   object 
 11  Star2          1000 non-null   object 
 12  Star3          1000 non-null   object 
 13  Star4          1000 non-null   object 
 14  No_of_Votes    1000 non-null   int64  
 15  Gross          831 non-null    object 
dtypes: float64(2), int64(1), object(13)
memory usage: 125.1+ KB
None


This shows the information about the movie dataset, it includes the data types of each column, the number of non-null values in each column, and memory usage.

The dataset has 1000 entries/rows and 16 columns
The memory ysage of the dataset is around 125.1 KB
The non-null count columns show the number of non-null values in each column, there seems to be around 899 non null values in the Certificate column. This would mean there are 101 missing values in the column.